### PR TITLE
Fungal Rebalance 1: JSONize, rebalance, and redraw fungal maps

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Move fungal tower map to JSON and redraw"```

#### Purpose of change
As one of the main factions of interdimensional horror, the fungus - while extremely compelling - remain so poorly balanced that many people just turn them off.

In a final push before 0.E, I would like to rebalance fungus, and improve their existing content to be more interesting.

#### Describe the solution
Fungal Towers should be the "capital cities" of the fungus. In upcoming updates I (ok I will probably ask someone else) would like to make fungification limited by proximity to a fungal tower. To have them be that important, they need to be a little interesting to encounter as well.

This is what they currently look like. It *does the trick*, but it's definitely not up to current mapping standards.
fungaltower_old

This is what they look like now:
fungaltower_new

Of note, I have added a "fungaloid queen" unit and a second tower. I imagine these three working together to represent the fungal core of the area, the queen producing spores and the towers linking the mycus to its interdimensional core.

I have also added a new furniture, fungal_tangle, a blocking terrain that looks like the fungal wall attack of the tower. It makes it a bit of a maze, potentially impassable without some smashing, before you can reach the gooey center of the special.

I have also added a physical attack to mon_fungal_wall at very low skill, as it blindly thrashes about with growing fungal tendrils - it mostly catches you by accident if you get too close.  I bumped its HP up a bit so it gets a little more time to smack you around.

#### Describe alternatives you've considered
I thought about making it a multiple z-level tall tower map with the mon_tower at the top, but that's a lot of work and I don't think it adds much to play: once you reach the area you're probably safe from most fungal threats, so spending longer chopping through them isn't much fun.

I've considered adding some fungal traps as well. Still worth considering, but I wouldn't mind testing this through a few rounds of players before going there.

#### Testing
I have tested this in game and it seems to run fine; however, without a compile environment I have not yet confirmed that the mapgen functions for hard coded fungal towers are correctly removed. I will enlist some help with that before this is ready to merge.

UPDATE: Code changes kindly tested by Fish Knight to ensure the old school fungal_tower isn't spawning and the game isn't crashing. Should be safe to merge.

#### Additional context
More to come.

### CODE REVIEW

**Comment by mkikt4743** [comment]
Seems very cool, and fungaloids are OP, def need a spread limit and a rebalance

**Comment by I-am-Erk** [comment]
I'm returning this to WIP for a few minutes, gonna just get fungal bloom and fungal flowers while I'm at it since it's pretty trivial.

File: data/json/furniture_and_terrain/furniture-fungal.json [hunk 1]
```diff
@@ -48,5 +48,18 @@
...
+    "type": "furniture",
+    "id": "f_fungal_tangle",
+    "name": "fungal tangle",
+    "description": "Thick, ropy tendrils of fungus have risen from the ground and gathered into an impenetrable clump.",
+    "looks_like": "mon_fungal_wall",
+    "symbol": "&",
+    "bgcolor": "light_gray",
+    "move_cost_mod": -1,
+    "required_str": -1,
+    "flags": [ "ALLOW_FIELD_EFFECT", "FLAMMABLE_ASH", "FUNGUS" ],
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/furniture-fungal.json]: Unless I'm reading code wrong, `ALLOW_FIELD_EFFECT` flag is meant to work only with or without the `SEALED` flag. Alone `ALLOW_FIELD_EFFECT` flag won't do anything useful.

**Comment by Torbenbeta** [comment]
Although I am no contributor, therefore not qualified to criticize, I would like to know. @I-am-Erk your rationale behind adding a fungal queen unit. I imagined that the units (Fungaloids) act semi indepently, needing no leader/queen and their reproduction mechanism needs no queen, being spore driven (instead of say egg-based like Bees). Furthermore it would increase their similarity to Triffids and mutant insects, reducing uniqueness and strangeness (A hierarchy is something we know and understand).

File: data/json/furniture_and_terrain/furniture-fungal.json (hunk 1 above)
Comment (I-am-Erk) [reply data/json/furniture_and_terrain/furniture-fungal.json]: Woops, thanks for spotting. I just forgot to delete it after copying the code from fungal clump

**Comment by I-am-Erk** [comment]
Fungal queens have existed for many years, torbenbeta, they're found in fungal blooms. I just added them to this map. They don't control fungaloids any more than a queen bee controls bees, they just dump out a lot of spores

**Comment by Xenomorph-III** [comment]
What does the RISK_HIGH flag do?

**Comment by anothersimulacrum** [comment]
It doesn't do anything now, but it's for NPCs.

**Pull request merged**